test/inductor/test_torchinductor_codegen_dynamic_shapes.py

```diff
@@ -193,7 +193,7 @@ def run(*ex, **kwargs):
     "test_new_empty_dynamic_shapes": TestFailure(("cpu", "cuda", "xpu")),
     "test_new_empty_strided_dynamic_shapes": TestFailure(("cpu", "cuda", "xpu")),
     "test_new_ones_dynamic_shapes": TestFailure(("cpu",)),
-    "test_permute2_dynamic_shapes": TestFailure(("cpu", "cuda", "xpu")),
+    "test_permute2_dynamic_shapes": TestFailure("xpu"),
```

Comment:
This is actually a bad thing, because we are not turning what used to have no triton kernels a triton copy.

Reply:
after I move all to a new API, they have expected failures now. let's wait for the final ci testings.

Reply:
The new API is slightly different from the previous function flow. will check the performance results for more details.

Reply:
here is the original output code:
```python
    return (reinterpret_tensor(arg1_1, ((-1) + s0, s0, 2, 1), (s0, 1, s0, s0), 0), )
```
here is after this PR:
```python
    buf0 = empty_strided_cpu(((-1) + s0, s0, 2, 1), (s0, 1, s0, s0), torch.float32)
    cpp_fused_permute_0(arg1_1, buf0, s0)
    del arg1_1
    return (buf0, )
```
new functions are generated but it makes the results correct. do you think it is the right thing?

Reply:
I'm not sure if the hud results is misleading. Because the compiled code with this PR for the model I tested is exact same as the base commit.

Reply:
It's not the same thing, because there is an additional kernel launched